nixos/modules/services/mail/zeyple.nix

```
@@ -0,0 +1,91 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.zeyple;
+  bool2int = x: if x then "1" else "0";
+
+  gpgHome = pkgs.runCommand "zeyple-gpg-home" { } ''
+    mkdir -p $out
+    for file in ${lib.concatStringsSep " " cfg.keys}; do
+      ${pkgs.gnupg}/bin/gpg --homedir="$out" --import "$file"
+    done
+
+    # Remove socket files
+    rm -f $out/S.gpg-agent*
+  '';
+in {
+  options.services.zeyple = {
+    enable = mkEnableOption "Zeyple, an utility program to automatically encrypt outgoing emails with GPG";
+
+    user = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "User to run Zeyple as.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "zeyple";
+      description = "Group to use to run Zeyple.";
+    };
+
+    logFile = mkOption {
+      type = types.path;
+      default = "/var/log/zeyple.log";
+      description = "Path of the log file.";
+    };
+
+    forceEncrypt = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to abort sending mail if no keys are found.";
+    };
+
+    keys = mkOption {
+      type = with types; listOf path;
+      description = "List of public key files that will be imported by gpg.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups."${cfg.group}" = { };
+    users.users."${cfg.user}" = {
+      isSystemUser = true;
+      group = "${cfg.group}";
+    };
```

It is a common pattern in NixOS modules to guard these based on whether or not the user has specified a customer user and group. You might be making too many assumptions otherwise.

```suggestion
    users.groups = optionalAttrs (cfg.group == "zeyple") {
      "${cfg.group}" = { };
    };
    users.users = optionalAttrs (cfg.user == "zeyple") {
      "${cfg.user}" = {
        isSystemUser = true;
        group = cfg.group;
      };
    };
```